mcode/base-icare-extraction-client
# Added allEntries cli option to indicate no date filtering of data

- Added `-a --all-entries` to indicate filtering of the data by date is not needed.
    - ~~Any suggestions on the naming of this flag? `-d` was already taken by debug. When I made it `--no-date-flag`, commander defaults the flag to `true` which can be handled but could be confusing in the code to read.~~
- Decided not to log the run when it was run with this flag since we have no `fromDate`. If we still want this run logged let me know and I can take that change out.

Testing:
When `allEntries` is provided, a `null` `fromDate` is passed to the client which is then passed down to the extractors. The `CSVModule` already has logic for handling a `null` `fromDate` so no changes were needed there. 
Follow the steps below to confirm the `allEntries` is working:

- Clear your `run-logs.json` and make sure it is an empty array.
- Run the client with this command `node src/cli.js -p src/csv.config.json -f 2020-01-01`
    - This should extract all the resources in our csvs.
- Run the client again without a `fromDate` `node src/cli.js -p src/csv.config.json`
    - No disease status nor treatment plan change(the two resources that have date filtering applied) should be extracted
- Run the client again with the `-n` flag `node src/cli.js -p src/csv.config.json -a`
    - Both disease status and treatment plan change should get extracted, ignoring the date filter.
